src/lightning/fabric/utilities/warnings.py

```diff
@@ -13,12 +13,22 @@
 # limitations under the License.
 """Warning-related utilities."""
 import warnings
+from typing import Optional, Type, Union
 
 from lightning.fabric.utilities.rank_zero import LightningDeprecationWarning
 
 # enable our warnings
 warnings.simplefilter("default", category=LightningDeprecationWarning)
 
 
+def _format_warning(
+    message: Union[Warning, str], category: Type[Warning], filename: str, lineno: int, line: Optional[str] = None
+) -> str:
+    return f"{filename}:{lineno}: {message}\n"
+
+
+warnings.formatwarning = _format_warning
```

Comment:
This will modify all warnings for all libs on import. We could instead only 
do this when the trainer/fabric starts or only apply the format when it's a warning of ours. The second option would be simpler

Reply:
I put in a function that checks whether the path of the file where the warning is emitted is a subpath of the lightning package. Took a while to figure out how to do this without making the CI explode.

Reply:
![image](https://github.com/Lightning-AI/lightning/assets/56649829/44e14fd8-0250-4d89-8ceb-387488b69532)